File: LeetCode/Python/Medium/2971-Find-Polygon-With-the-Largest-Perimeter/solution.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        result=-1

        n=len(nums)

        nums.sort()

        prefix_sums=[0 for _ in range(0, n)]

        prefix_sums[0]=nums[0]

        for index in range(1, n):
            prefix_sums[index]=nums[index] + prefix_sums[index - 1]

        for index in range(2, n):
            largest_side=prefix_sums[index] - prefix_sums[index - 1]

            sum_of_remaining_sides=prefix_sums[index - 1]

            if largest_side < sum_of_remaining_sides:
                result=max(result, prefix_sums[index])

        return result

File: LeetCode/Rust/Medium/2971-Find-Polygon-With-the-Largest-Perimeter/solution.rs
impl Solution
{
    pub fn largest_perimeter(mut nums: Vec<i32>) -> i64
    {
        let mut result: i64=-1;

        let n: usize=nums.len();

        nums.sort();

        let mut prefix_sums: Vec<i64>=vec![0; n];

        prefix_sums[0]=nums[0] as i64;

        for index in 1..n
        {
            prefix_sums[index]=prefix_sums[index - 1] + (nums[index] as i64);
        }

        for index in 2..n
        {
            let largest_side: i64=prefix_sums[index] - prefix_sums[index - 1];

            let sum_of_remaining_sides: i64=prefix_sums[index - 1];

            if largest_side < sum_of_remaining_sides
            {
                result=std::cmp::max(result, prefix_sums[index]);
            }
        }

        result
    }
}